### DISCUSSION TITLE
Creating a dynamic/real-time voxelgrid visualization

### DISCUSSION THREAD

**Discussion by whizbuzzer** [isl-org/Open3D#3829] [Q&A]
I am using Intel RealSense D455 for point cloud visualization.

I have managed to create a real-timepoint cloud visualization with `add_geometry()` and `update_geometry()` functions, but these functions are not functioning as I want them to for voxel grids; the voxel grid is indeed created and visualized, but it does not move with me the same way that the point cloud does. Does anyone know a fix to this?

**Comment by germanros1987** [comment]
We added this to our feature request list.

**Reply by saurabheights** [reply]
A hack solution for this is:-

```
prev_voxel_grid = None

### Compute Voxel grid ### 

if prev_voxel_grid is not None:
    vis.remove_geometry(prev_voxel_grid, False)
prev_voxel_grid = voxel_grid
vis.add_geometry(voxel_grid, False)
```
